boards/risc-v/bl602/bl602evb/scripts/Make.defs

```diff
@@ -45,8 +45,8 @@ ifneq ($(CONFIG_DEBUG_NOOPT),y)
 endif
 
 ARCHCPUFLAGS += -mno-relax
-ARCHCFLAGS = -fno-builtin -ffunction-sections -fdata-sections -fstrict-volatile-bitfields -fshort-enums -fno-omit-frame-pointer
-ARCHCXXFLAGS = -fno-builtin -ffunction-sections -fdata-sections -fstrict-volatile-bitfields -fshort-enums -fno-omit-frame-pointer -fno-exceptions -fcheck-new -std=c++17 -D__NuttX__ -pipe -nostdinc++
+ARCHCFLAGS = -fno-builtin -ffunction-sections -fdata-sections
```

Comment:
General question on `-fshort-enums`. Shouldn't this give better code size?
Do we have a Kconfig options to specify extra options like this? It might be needed if code is linked to third party pre-compiled libraries that are compiled with `-fshort-enums` and have pointers to enums or arrays of enums passed as parameters to functions.

Reply:
> General question on `-fshort-enums`. Shouldn't this give better code size? 

Since the prebuilt library(e.g. libgcc, libm, libsupc++) from toolchain never add -fshort-enums, it's wrong to enable this option unless we build the toolchain by self.

> Do we have a Kconfig options to specify extra options like this? It might be needed if code is linked to third party pre-compiled libraries that are compiled with `-fshort-enums` and have pointers to enums or arrays of enums passed as parameters to functions.

No.

Reply:
@xiaoxiang781216 if I recall this was required for abi compatibility when calling some ROM functions which unfortunately used enums as part of the API.  I know this is the case also with the Nordic Bluetooth softcore as well.

Reply:
It isn't a big issue If ROM function doesn't use array argument, since the type will always promote to int per spec.

Reply:
Probably we need to check ROM API calls if those have enum arrays as arguments or return values.